packages/devtools_app/lib/src/screens/deep_link_validation/deep_link_list_view.dart

```diff
@@ -81,78 +79,102 @@ class _DeepLinkListViewMainPanel extends StatelessWidget {
+
+    return ValueListenableBuilder<PagePhase>(
+      valueListenable: controller.pagePhase,
+      builder: (context, pagePhase, _) {
+        switch (pagePhase) {
+          case PagePhase.emptyState:
+          case PagePhase.linksLoading:
+          case PagePhase.linksValidating:
             return Column(
               mainAxisAlignment: MainAxisAlignment.center,
               children: [
                 const CenteredCircularProgressIndicator(),
                 const SizedBox(height: densePadding),
                 Text(
-                  'Validating deep links...',
+                  pagePhase == PagePhase.linksLoading
+                      ? 'Loading deep links...'
+                      : 'Validating deep links...',
                   style: Theme.of(context).subtleTextStyle,
                 ),
               ],
             );
-          }
-          if (displayOptions.showSplitScreen) {
-            return Row(
-              children: [
-                Expanded(
-                  child: _AllDeepLinkDataTable(controller: controller),
-                ),
-                VerticalDivider(
-                  width: 1.0,
-                  color: Theme.of(context).focusColor,
-                ),
-                Expanded(
-                  child: ValueListenableBuilder<LinkData?>(
-                    valueListenable: controller.selectedLink,
-                    builder: (context, selectedLink, _) => TabBarView(
-                      children: [
-                        ValidationDetailView(
-                          linkData: selectedLink!,
-                          controller: controller,
-                          viewType: TableViewType.domainView,
-                        ),
-                        ValidationDetailView(
-                          linkData: selectedLink,
-                          controller: controller,
-                          viewType: TableViewType.pathView,
-                        ),
-                        ValidationDetailView(
-                          linkData: selectedLink,
-                          controller: controller,
-                          viewType: TableViewType.singleUrlView,
-                        ),
-                      ],
-                    ),
-                  ),
-                ),
-              ],
-            );
-          }
-          return Column(
-            crossAxisAlignment: CrossAxisAlignment.start,
+          case PagePhase.linksValidated:
+            return const _VerifiedDeepLinksView();
+          case PagePhase.noLinks:
+            // TODO(hangyujin): This is just a place holder to add UI.
+            return const Text('Your flutter project has no Links to verify.');
+
+          case PagePhase.errorPage:
+            // TODO(hangyujin): This is just a place holder to add Error handling.
+            return const Text('Error');
+        }
+      },
+    );
+  }
+}
+
+class _VerifiedDeepLinksView extends StatelessWidget {
```

Comment:
nit: should this be named `_ValidatedDeepLinksView` to match other "validate" terminology?